chainer/functions/evaluation/r2_score.py

```diff
@@ -0,0 +1,70 @@
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class R2_score(function.Function):
+
+    def __init__(self, sample_weight, multioutput):
+        if sample_weight is not None:
+            raise NotImplementedError()
+        if multioutput in ['uniform_average', 'raw_values']:
+            self.multioutput = multioutput
+        else:
+            raise ValueError("invalid multioutput argument")
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        pred_type, true_type = in_types
+
+        type_check.expect(
+            pred_type.dtype.kind == 'f',
+            true_type.dtype.kind == 'f'
+        )
+
+        type_check.expect(
+            pred_type.shape == true_type.shape,
+        )
+
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
+        pred, true = inputs
+        SS_res = xp.sum((pred - true) ** 2, axis=0)
+        SS_tot = xp.sum((true - xp.mean(true, axis=0)) ** 2, axis=0)
+        if self.multioutput == 'uniform_average':
+            if xp.any(SS_tot == 0):
+                return xp.asarray(0.0, dtype=pred.dtype),
+            else:
+                return xp.asarray((1 - SS_res / SS_tot).mean(),
+                                  dtype=pred.dtype),
+        elif self.multioutput == 'raw_values':
+            if xp.any(SS_tot == 0):
```

Comment:
If the reason for this branching is to avoid zero division by `SS_tot`, we cannot avoid it because `SS_res / SS_tot` is evaluated anyway in case `SS_tot == 0`. Are there other reason for this branching? If not, I recommend to remove the branching. By removing it, we can expect faster forward computation.